### PR TITLE
Add version endpoint

### PR DESCRIPTION
**Pull request by arijitAD**
## Motivation and Context

- [ ] I have raised an issue to propose this change ([required](https://github.com/ovrclk/akash/blob/master/CONTRIBUTING.md))
- [ ] My issue has received approval from the maintainers or lead with the `design/approved` label

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I've read the [CONTRIBUTION](https://github.com/ovrclk/akash/blob/master/CONTRIBUTING.md) guide
- [ ] I have signed-off my commits with `git commit -s`
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: provider/cluster/kube/client.go [hunk 1]
```diff
@@ -786,3 +787,7 @@ func (c *client) deploymentsForLease(ctx context.Context, lid mtypes.LeaseID) (m
 
 	return serviceStatus, nil
 }
+
+func (c *client) KubeVersion() (*version.Info, error) {
+	return c.kc.(*kubernetes.Clientset).ServerVersion()
```
Comment (hydrogen18) [inline provider/cluster/kube/client.go:792]: It seems like this method exists on an interface on its own. It's more robust to just cast to that interface type. Also we should check for the cast failing and just return an error.

https://github.com/kubernetes/client-go/blob/master/discovery/discovery_client.go#L120

Comment (boz) [reply provider/cluster/kube/client.go:792]: Yes, always use methods where available.

Take the time to do things right @arijitAD

**Review by boz — CHANGES_REQUESTED** [review]

File: provider/cluster/kube/client.go (hunk 1 above)
Comment (arijitAD) [reply provider/cluster/kube/client.go]: Fixed

File: provider/gateway/rest/router_test.go [hunk 2]
```diff
@@ -307,6 +308,44 @@ func TestRouteDoesNotExist(t *testing.T) {
 	})
 }
 
+func TestRouteVersionOK(t *testing.T) {
+	runRouterTest(t, false, func(test *routerTest) {
+		status := versionInfo{
+			Akash: "",
```
Comment (hydrogen18) [inline provider/gateway/rest/router_test.go:314]: Please use a non-empty string here for the test. Anything is fine, just `"akashtest"` or whatever should be OK

**Comment by hydrogen18** [comment]
Just the one comment about the test that should be improved, other than that this looks good to me.

**Review by hydrogen18 — APPROVED** [review]

File: provider/gateway/rest/router.go [hunk 3]
```diff
@@ -391,6 +400,25 @@ func leaseShellHandler(log log.Logger, mclient pmanifest.Client, cclient cluster
 	}
 }
 
+type versionInfo struct {
+	Akash string            `json:"akash"`
```
Comment (boz) [inline provider/gateway/rest/router.go:404]: should we include commit hash, build date, go version, platform, etc...?

File: provider/gateway/rest/router_test.go (hunk 2 above)
Comment (arijitAD) [reply provider/gateway/rest/router_test.go]: Done

File: provider/gateway/rest/router.go (hunk 3 above)
Comment (arijitAD) [reply provider/gateway/rest/router.go]: Added all the useful information that is available via `akash version` command.

**Review by boz — APPROVED** [review]
:rocket:

**Pull request merged**